docs/proposals/20191017-control-plane.md

```diff
@@ -0,0 +1,495 @@
+---
+title: Kubeadm Machine Control Plane Management  
+authors:
+  - "@detiber”
+  - "@chuckha”
+  - "@randomvariable"
+  - "@dlipovetsky"
+reviewers:
+  - "@ncdc"
+  - "@timothysc"
+  - "@vincepri"
+creation-date: 2019-09-18
+last-updated: 2019-09-18
+status: provisional
+---
+
+# Kubeadm Machine Control Plane Management
+
+## Table of Contents
+
+- [Kubeadm Machine Control Plane Management](#kubeadm-machine-control-plane-management)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+    - [References](#references)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+      - [Additional goals of the default kubeadm machine-based Implementation](#additional-goals-of-the-default-kubeadm-machine-based-implementation)
+    - [Non-Goals / Future Work](#non-goals--future-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Identified features from user stories](#identified-features-from-user-stories)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      - [New API Types](#new-api-types)
+      - [Modifications required to existing API Types](#modifications-required-to-existing-api-types)
+      - [Behavioral Changes from v1alpha2](#behavioral-changes-from-v1alpha2)
+      - [Behaviors](#behaviors)
+        - [Create](#create)
+        - [Scale Up](#scale-up)
+        - [Scale Down](#scale-down)
+        - [Delete of the entire ControlPlane (kubectl delete controlplane my-controlplane)](#delete-of-the-entire-controlplane-kubectl-delete-controlplane-my-controlplane)
+        - [Cluster upgrade (using create-swap-and-delete)](#cluster-upgrade-using-create-swap-and-delete)
+        - [Control plane healthcheck](#control-plane-healthcheck)
+        - [Adoption of pre-v1alpha3 Control Plane Machines](#adoption-of-pre-v1alpha3-control-plane-machines)
+      - [Code organization](#code-organization)
+    - [Risks and Mitigations](#risks-and-mitigations)
+      - [etcd membership](#etcd-membership)
+  - [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Alpha -> Beta Graduation](#alpha---beta-graduation)
+    - [Upgrade Strategy](#upgrade-strategy)
+  - [Alternatives](#alternatives)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+The lexicon used in this document is described in more detail [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/glossary.md).  Any discrepancies should be rectified in the main Cluster API glossary.
+Stacked Control Plane / Stacked etcd: Control plane node where etcd is colocated with the Kubernetes API server, and is running as a static pod.
+
+### References
+
+[Kubernetes Control Plane Management Overview](https://docs.google.com/document/d/1nlWKEr9OP3IeZO5W2cMd3A55ZXLOXcnc6dFu0K9_bms/edit#)
+
+## Summary
+
```

Comment:
Suggest adding an intro, something like:

This proposal outlines adding control plane management to Cluster API....